### DISCUSSION TITLE
Extract the number of in-flight requests?

### DISCUSSION THREAD

**Discussion by deliahu** [fastapi/fastapi#7796] [Questions]
### First check

* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.

### Description

Is there a way to extract the total number of in-flight requests? I.e. the number of requests that have been received and not yet responded to (not including background tasks).

If so, how can it be consumed? E.g. can I create an endpoint like `/requests` which responds with the number? Or perhaps a cron task which can query the number so I could send it to a stats database?

### Motivation

I'd like to trigger autoscaling events based on the number of in-flight requests.

**Comment by phy25** [comment]
You can look for some Starlette solutions probably e.g. https://github.com/perdy/starlette-prometheus

**Comment by deliahu** [comment]
Yes, I will look into using Starlette middleware, thank you!

**Comment by tiangolo** [comment]
Thanks for the help here @phy25 ! :cake: :bowing_man: 

Thanks @deliahu for reporting back and closing the issue :+1:
